infrastructure/tooling/src/importer/importer.js

```diff
@@ -19,67 +20,116 @@ const importer = async options => {
+  }
 
-        async function* readAzureTableIterator(args) {
-          yield await readAzureTableInChunks(args);
+  async function importTable(tableName, tableParameters = {}, rowsProcessed = 0, utils) {
+    for await (let result of buffer(
+      readAzureTableIterator({
+        azureCreds: credentials.azure,
+        tableName,
+        utils,
+        tableParams: tableParameters,
+        rowsProcessed,
+      }),
+      1000,
+    )) {
+      if (result) {
+        const { entities, tableParams, count } = result;
+
+        await writeToPostgres(tableName, entities, db, ALLOWED_TABLES);
+
+        if (tableParams.nextPartitionKey && tableParams.nextRowKey) {
+          return await importTable(tableName, tableParams, count, utils);
         }
 
-        async function importTable(tableParameters = {}, rowsProcessed = 0) {
-          for await (let result of buffer(
-            readAzureTableIterator({
-              azureCreds: credentials.azure,
-              tableName,
-              utils,
-              tableParams: tableParameters,
-              rowsProcessed,
-            }),
-            1000,
-          )) {
-            if (result) {
-              const { entities, tableParams, count } = result;
-
-              await writeToPostgres(tableName, entities, db, ALLOWED_TABLES);
-
-              if (tableParams.nextPartitionKey && tableParams.nextRowKey) {
-                return await importTable(tableParams, count);
-              }
-
-              return count;
-            }
-
-            return 0;
-          }
-        }
+        return count;
+      }
 
-        const rowsImported = await importTable();
+      return 0;
+    }
+  }
 
-        return {
-          [tableName]: {
-            elapsedTime: new Date() - start,
-            rowsImported,
+  let largeTableNames = [];
+  for (let tableName of tables) {
+    if (LARGE_TABLES.includes(tableName)) {
+      const pgTable = postgresTableName(tableName);
+
+      await db._withClient('admin', async client => {
+        await client.query(`truncate ${pgTable}`);
+      });
+      const january2021 = new Date(2021, 0, 1);
+      const june2021 = new Date(2021, 6, 1);
+
+      [
+        {
+          name: `${tableName}-1/3`,
+          filter: `expires ${Table.Operators.LessThan} ${Table.Operators.date(january2021)}`,
+          title: `expires < ${january2021.toJSON()}`,
+        },
+        {
+          name: `${tableName}-2/3`,
+          filter: `expires ${Table.Operators.GreaterThanOrEqual} ${Table.Operators.date(january2021)} ${Table.Operators.And} expires ${Table.Operators.LessThan} ${Table.Operators.date(june2021)}`,
+          title: `expires >= ${january2021.toJSON()} and < ${june2021.toJSON()}`,
+        },
+        {
+          name: `${tableName}-3/3`,
+          filter: `expires ${Table.Operators.GreaterThanOrEqual} ${Table.Operators.date(june2021)}`,
+          title: `expires >= ${june2021.toJSON()}`,
+        },
```

Comment:
@imbstack In an effort to make the importer run faster, does this logic make sense to you?